packages/fxa-settings/src/models/Account.ts

```diff
@@ -623,10 +623,41 @@ export class Account implements AccountData {
     return result;
   }
 
+  /**
+   * TODO: Remove this once we update the GQL endpoint to accept the
+   *  `accountResetWithRecoveryKey` option and fix graphql-api not reporting
+   *  the correct IP address.
+   *
+   * @param token passwordForgotToken
+   * @param code code
+   * @param accountResetWithRecoveryKey is account being reset with recovery key?
+   * */
+  async passwordForgotVerifyCode(
+    token: string,
+    code: string,
+    accountResetWithRecoveryKey = false
+  ): Promise<string> {
+    // TODO: There is a bug in Backbone and React reset PW around `accountResetWithRecoveryKey`.
+    // We attempt to validate the `code` and `token` provided here, but because the
+    // `accountResetToken` can only be fetched once but uses the `accountResetWithRecoveryKey`
+    // option, auth-server fails to send an email if the user has a recovery key, attempts
+    // to use it unsuccessfully, and then goes through a normal reset via the link back
+    // to a normal reset if a user can't use their key.
```

Comment:
I will let QA know about this. Since it's been an issue for a while I'm not sure we'll want to address it before the password reset redesign.

Reply:
Good catch, I've confirmed I see the same issue, with no email sent out if the user tries an incorrect recovery key then resets the password without the key. I don't _think_ this is a prod blocker since it's been an issue for a while, but would be good to mention to UX for the redesign.